nautobot/docs/development/best-practices.md

```diff
@@ -63,3 +63,193 @@ class ExampleModel(PrimaryModel):
+                 },
+             },
+             models.BooleanField: {
+                 'filter_class': django_filters.BooleanFilter,
+                 'extra': lambda f: {
+                     'widget': forms.CheckboxInput,
+                 },
+             },
+         }
+```
+
+!!!warning
+    Existing features of filtersets and filters **must** be exhausted first using keyword arguments before resorting to customizing, re-declaring/overloading, or defining filter methods.
+
+### Filter Methods
+
+Filter methods in the current Nautobot core are problematic because they break the ability for such filter fields to be properly reversible. Specifically a filter method is a callable or the name of a method on a filterset that can be declared on a filter when it is defined on a filterset. This method is used to perform custom business logic on a filter field.
+
+Consider this example from `nautobot.dcim.filters.DeviceFilterSet`:
+
+```python
+  # Filter field definition is a BooleanFilter, for which an "isnull" lookup_expr 
+  # is the only valid filter expression
+  console_ports = django_filters.BooleanFilter(
+      method="_console_ports",
+      label="Has console ports",
+  )
+  
+  # Method definition loses context and further the field's lookup_expr 
+  # falls back to the default of "exact".
+  def _console_ports(self, queryset, name, value):
+      breakpoint()  # This is where we'll illustrate pdb below
+      return queryset.exclude(consoleports__isnull=value)
+```
+
+The default `lookup_expr` unless otherwise specified is “exact”, as seen in [django_filters.conf](https://github.com/carltongibson/django-filter/blob/main/django_filters/conf.py#L10):
+
+```python
+  'DEFAULT_LOOKUP_EXPR': 'exact',
+```
+
+When this method is called, the internal state is default, making reverse introspection impossible, because the `lookup_expr` is defaulting to “exact”:
+
+```python
+(Pdb) field = self.filters["console_ports"]
+(Pdb) field.exclude
+False
+(Pdb) field.lookup_expr
+'exact' 
+```
+
+This means that the arguments for the field are being completely ignored and the hard-coded queryset `queryset.exclude(consoleports__isnull=value)` is all that is being run when this method is called. This hard-coding is impossible to introspect and therefore reverse.
+
+In fact, this method is completely unnecessary, because it could be replaced with options on the filter field itself:
+
+```python
+    console_ports = django_filters.BooleanFilter(
+        field_name="consoleports",  # The actual related field name
+        exclude=True,               # Perform an `.exclude()` vs. `.filter()``
+        lookup_expr="isnull",       # Perform `isnull` vs. `exact``
+        label="Has console ports",
+    )
+```
+
+Now, if we use another breakpoint (this time inside of the `DeviceListView.get()` since there’s no method we can break into), you can see that the filter field now has the correect attributes that can be used to accurately reverse this query:
```

Comment:
```suggestion
Now, if we use another breakpoint (this time inside of the `DeviceListView.get()` since there’s no method we can break into), you can see that the filter field now has the correct attributes that can be used to accurately reverse this query:
```